src/Common/Notifications/Notifications.php

```diff
@@ -0,0 +1,347 @@
+<?php
+/**
+ * Handles In-App Notifications setup and actions.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Notifications
+ */
+
+namespace TEC\Common\Notifications;
+
+use Tribe__Main;
+use TEC\Common\Admin\Conditional_Content\Dismissible_Trait;
+
+/**
+ * Class Notifications
+ *
+ * @since   TBD
+
+ * @package TEC\Common\Notifications
+ */
+final class Notifications {
+	use Dismissible_Trait;
+
+	/**
+	 * The slugs for plugins that support In-App Notifications.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	private $slugs = [];
+
+	/**
+	 * The Notifications API URL.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	private $api_url = '';
+
+	/**
+	 * Notification slug for dismissible content.
+	 *
+	 * @var string
+	 */
+	protected string $slug = '';
+
+	/**
+	 * Boot the in-App Notifications
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function boot(): void {
+		$this->api_url = $this->get_api_url();
+		$this->slugs   = $this->get_plugins();
+
+		/**
+		 * Allow plugins to hook in and add themselves,
+		 * running their own actions once In-App Notifications is initiated.
+		 *
+		 * @since TBD
+		 *
+		 * @param self $ian The In-App Notifications instance.
+		 */
+		do_action( 'tec_common_ian_preload', $this );
+	}
+
+	/**
+	 * Initializes the plugins and triggers the "loaded" action.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	public function init(): void {
+		/**
+		 * Allow plugins to hook in and add themselves,
+		 * running their own actions once IAN is initiated.
+		 *
+		 * @since TBD
+		 *
+		 * @param self $ian The IAN instance.
+		 */
+		do_action( 'tec_common_ian_loaded', $this );
+	}
+
+	/**
+	 * Register the Admin assets for the In-App Notifications.
+	 *
+	 * @since  TBD
+	 *
+	 * @return void
+	 */
+	public function register_assets(): void {
+		tribe_assets(
+			Tribe__Main::instance(),
+			[
+				[ 'ian-client-css', 'ian-client.css' ],
+				[ 'ian-client-js', 'ian-client.js', [ 'jquery' ] ],
+			],
+			'admin_enqueue_scripts',
+			[
+				'conditionals' => [ $this, 'is_ian_page' ],
+				'in_footer'    => false,
+				'localize'     => [
+					'name' => 'commonIan',
+					'data' => [
+						'ajax_url' => admin_url( 'admin-ajax.php' ),
+						'nonce'    => wp_create_nonce( 'common_ian_nonce' ),
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get the API URL for the In-App Notifications.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_api_url() {
+		$api = 'https://ian.stellarwp.com/feed/organization/brand/product.json';
+
+		/**
+		 * Filter the API URL for the In-App Notifications.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $api The API URL for the In-App Notifications.
+		 */
+		$api = apply_filters( 'tec_common_ian_api_url', $api );
```

Comment:
Can we also add some parameters to this filter so someone would have more information on the source of the call? For instance trying to see the product calling it? This might not be necessary if we decide to implement the above.

Reply:
Indeed, I believe @Camwyn told me before to always pass `$this` as the last param because it will probably be needed down the road and to add context to the filter. Thanks for pointing it out, I will change that.